fml/message_loop_impl.cc

```diff
@@ -119,8 +123,18 @@ void MessageLoopImpl::SwapTaskQueues(const fml::RefPtr<MessageLoopImpl>& other)
 
   std::lock(t1, t2, o1, o2, d1, d2);
 
-  std::swap(task_observers_, other->task_observers_);
-  std::swap(delayed_tasks_, other->delayed_tasks_);
+  // inherit all the task_observers.
```

Comment:
We wont ever be able to unmerge without also splitting observers correctly. Lets just fail in this case.